app.py

```diff
@@ -670,53 +597,88 @@ def ubi(statefip, level, agi_tax, benefits, taxes, exclude):
             target_spmu.new_resources / target_spmu.numper
         )
 
-    # Merge and create target_persons
+    # Merge and create target_persons -
+    # NOTE: the "target" here refers to the population being
+    # measured for gini/poverty rate/etc.
+    # I.e. the total population of the state/country and
+    # INCLUDING those excluding form recieving ubi payments
     sub_spmu = target_spmu[
         ["spmfamunit", "year", "new_resources", "new_resources_per_person"]
     ]
     target_persons = person.merge(sub_spmu, on=["spmfamunit", "year"])
 
-    # Calculate populations
-    population = target_persons.asecwt.sum()
-    child_population = (target_persons.child * target_persons.asecwt).sum()
-    non_citizen_population = (
-        target_persons.non_citizen * target_persons.asecwt
-    ).sum()
-    non_citizen_child_population = (
-        target_persons.non_citizen_child * target_persons.asecwt
-    ).sum()
+    # filter demog_stats for selected state from dropdown
+    baseline_demog = demog_stats[(demog_stats.index == state)]
+
+    def return_demog(demog, metric, pct=True):
+        """
+        retrieve pre-processed data by demographic
+        args:
+            demog - string one of
+                ['person', 'adult', 'child', 'black', 'white_non_hispanic',
+            'hispanic', 'pwd', 'non_citizen', 'non_citizen_adult',
+            'non_citizen_child']
+            metric - string, one of ['pov_rate', 'pop']
+        returns:
+            value - float
+        """
+        df_copied = baseline_demog.copy(deep=True)
+
+        value = df_copied.loc[
+            (df_copied["demog"] == demog) & (df_copied["metric"] == metric),
+            "value",
+            # NOTE: returns the first value as a float, be careful if you redefine baseline_demog
+        ].values[0]
+
+        # multiply poverty rate by 100 to get percentage
+
+        return value
+
+    population = return_demog(demog="person", metric="pop")
+    child_population = return_demog(demog="child", metric="pop")
+    non_citizen_population = return_demog(demog="non_citizen", metric="pop")
+    non_citizen_child_population = return_demog(demog="non_citizen_child", metric="pop")
+
+    # filter all state stats gini, poverty_gap, etc. for dropdown state
+    baseline_all_state_stats = all_state_stats[(all_state_stats.index == state)]
+
+    def return_all_state(metric):
+        """filter baseline_all_state_stats and return value of select metric
+
+        Keyword arguments:
+        metric - string, one of 'poverty_gap', 'gini', 'total_resources'
+
+        returns:
+            value- float
+        """
+        df_copied = baseline_all_state_stats.copy(deep=True)
```

Comment:
similarly you can do this instead here
```suggestion
        return baseline_all_state_stats[metric].values[0]
```